code/modules/projectiles/projectile.dm

```diff
@@ -452,7 +452,7 @@ var/list/impact_master = list()
 /obj/item/projectile/proc/OnFired(var/proj_target = original)	//if assigned, allows for code when the projectile gets fired
 	target = get_turf(proj_target)
 
-	if(!original && !target)
+	if(isnull(target) || (!original && !target))
```

Comment:
```suggestion
	if(!target)
```
There's no point to the original check with this change, since it could never pass without the new check also passing.
I wonder if there was a reason it checked both.

Reply:
Good point, I will change this. However, I will err on the side of caution and instead change it to:
`if(!original || !target)`, per Pomf's suggestion, (instead of `&&` as it is now) in case there's good reason for it to check both.